arangod/Cluster/TraverserEngine.cpp

```diff
@@ -189,20 +189,19 @@ void BaseEngine::getVertexData(VPackSlice vertex, VPackBuilder& builder,
       return;
     }
     std::string_view vertex = id.substr(pos + 1);
-    auto cb = IndexIterator::makeDocumentCallbackF(
-        [&](LocalDocumentId, VPackSlice doc) {
-          // FOUND. short circuit.
-          read++;
-          builder.add(v);
-          if (!options().getVertexProjections().empty()) {
-            VPackObjectBuilder guard(&builder);
-            options().getVertexProjections().toVelocyPackFromDocument(
-                builder, doc, _trx.get());
-          } else {
-            builder.add(doc);
-          }
-          return true;
-        });
+    auto cb = [&](LocalDocumentId, aql::DocumentData&&, VPackSlice doc) {
+      // FOUND. short circuit.
+      read++;
+      builder.add(v);
+      if (!options().getVertexProjections().empty()) {
+        VPackObjectBuilder guard(&builder);
+        options().getVertexProjections().toVelocyPackFromDocument(builder, doc,
```

Comment:
what if `DocumentData` was set?

Reply:
We don't use DocumentData in case if some projection exist, I don't think it's good, but it's current state.

On my opinion best approach what we can to achieve is having multiple aql registers:
one with whole document
multiple with projected fields (just reference to whole document)
Note: whole document won't be serialized and sent via network

But such approach needs additional work.